staging/src/k8s.io/kube-aggregator/pkg/controllers/status/remote/remote_available_controller.go

```diff
@@ -199,21 +195,12 @@ func (c *AvailableConditionController) sync(key string) error {
 		},
 	}
 
-	apiService := originalAPIService.DeepCopy()
-
 	availableCondition := apiregistrationv1.APIServiceCondition{
 		Type:               apiregistrationv1.Available,
 		Status:             apiregistrationv1.ConditionTrue,
 		LastTransitionTime: metav1.Now(),
 	}
 
-	// local API services are always considered available
-	if apiService.Spec.Service == nil {
-		apiregistrationv1apihelper.SetAPIServiceCondition(apiService, apiregistrationv1apihelper.NewLocalAvailableAPIServiceCondition())
-		_, err := c.updateAPIServiceStatus(originalAPIService, apiService)
-		return err
-	}
```

Comment:
@aojea this change drove me crazy. I am moving the if condition above the rest client creation. It failed integration tests. Then I moved it up line-by-line, eventually realizing that `restTransport, err := transport.New(transportConfig)` is the critical line. WTF. That was the moment I questions my career choices and the universe. Eventually, I read `TestAPIServerTransportMetrics` 😀